crawler/crawler.go

```diff
@@ -146,3 +159,57 @@ func validateRemoteFile(data []byte, url string) error {
 
 	return publiccode.Parse(data, &pc)
 }
+
+// saveES save the chosen <file_name> in elasticsearch
+func saveToES(source, name string, data []byte) {
+	const (
+		index = "publiccode" // Elasticsearch index.
+		// Elasticsearch mapping for publiccode. Checkout elasticsearch/mappings/software.json
+		// TODO: Mapping must reflect the publiccode.PublicCode structure.
+		mapping = ""
+	)
+	// Starting with elastic.v5, you must pass a context to execute each service.
+	ctx := context.Background()
+
+	// Create a client.
+	client, err := elastic.NewClient(
+		elastic.SetURL(os.Getenv("ELASTIC_URL")),
+		elastic.SetSniff(false),
+		elastic.SetBasicAuth(os.Getenv("ELASTIC_USER"), os.Getenv("ELASTIC_PWD")))
+	if err != nil {
+		log.Error(err)
```

Comment:
maybe errors in connecting and creating index in ES should panic?

Reply:
Yes. Now ES is pretty slow to start (20-30 seconds in order to setup Guard) and I think it will be panic if launched at the same time of crawler. I'll search for a connect-retry function in ES package.